packages/editor-ui/src/components/WorkflowActivator.vue

```diff
@@ -74,59 +81,47 @@ export default mixins(
 					}
 					return '#13ce66';
 				},
+				isCurrentWorkflow(): boolean {
+					return this.$store.getters['workflowId'] === this.workflowId;
+				},
+				disabled(): boolean {
+					const isNewWorkflow = !this.workflowId;
+					if (isNewWorkflow || this.isCurrentWorkflow) {
+						return !this.workflowActive && !this.containsTrigger;
+					}
+
+					return false;
+				},
+				containsTrigger(): boolean {
+					const foundTriggers = getActivatableTriggerNodes(this.$store.getters.workflowTriggerNodes);
+					return foundTriggers.length > 0;
+				},
 			},
 			methods: {
 				async activeChanged (newActiveState: boolean) {
-					if (this.workflowId === undefined) {
-						this.$showMessage({
-							title: this.$locale.baseText('workflowActivator.showMessage.activeChangedWorkflowIdUndefined.title'),
-							message: this.$locale.baseText('workflowActivator.showMessage.activeChangedWorkflowIdUndefined.message'),
-							type: 'error',
-						});
-						return;
-					}
-
-					if (this.nodesIssuesExist === true) {
-						this.$showMessage({
-							title: this.$locale.baseText('workflowActivator.showMessage.activeChangedNodesIssuesExistTrue.title'),
-							message: this.$locale.baseText('workflowActivator.showMessage.activeChangedNodesIssuesExistTrue.message'),
-							type: 'error',
-						});
-						return;
-					}
+					this.loading = true;
 
-					// Set that the active state should be changed
-					let data: IWorkflowDataUpdate = {};
-
-					const activeWorkflowId = this.$store.getters.workflowId;
-					if (newActiveState === true && this.workflowId === activeWorkflowId) {
-						// If the currently active workflow gets activated save the whole
-						// workflow. If that would not happen then it could be quite confusing
-						// for people because it would activate a different version of the workflow
-						// than the one they can currently see.
-						if (this.dirtyState) {
-							const importConfirm = await this.confirmMessage(
-								this.$locale.baseText('workflowActivator.confirmMessage.message'),
-								this.$locale.baseText('workflowActivator.confirmMessage.headline'),
-								'warning',
-								this.$locale.baseText('workflowActivator.confirmMessage.confirmButtonText'),
-								this.$locale.baseText('workflowActivator.confirmMessage.cancelButtonText'),
-							);
-							if (importConfirm === false) {
-								return;
-							}
+					if (!this.workflowId) {
+						const saved = await this.saveCurrentWorkflow();
```

Comment:
ideally this would be part of the same request as below.. I tried. but too much work to refactor now for this. instead, we have two api requests, first to save, then to save again and update active state.